File: src/main/java/com/example/temperatura/converter/controller/TemperaturaConverterController.java
package com.example.temperatura.converter.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.temperatura.converter.interfaces.CalculadoraTemperatura;

@RestController
@RequestMapping("/converter")
public class TemperaturaConverterController {
	private final CalculadoraTemperatura calculadoraTemperatura;

	@Autowired
	public TemperaturaConverterController(CalculadoraTemperatura calculadoraTemperatura) {
		this.calculadoraTemperatura = calculadoraTemperatura;
	}

	@GetMapping("/ctof/{tempCelsius}")
	public Double celsiusToFareinheit(@PathVariable(name = "tempCelsius") Double tempCelsius) {
		return calculadoraTemperatura.celsiusToFarenheit(tempCelsius);
	}
	
	@GetMapping("/ctok/{tempCelsius}")
	public Double celsiusToKelvin(@PathVariable(name = "tempCelsius") Double tempCelsius) {
		return calculadoraTemperatura.celsiusToKelvin(tempCelsius);
	}
	
	@GetMapping("/ftoc/{tempFarenheit}")
	public Double farenheitToCelsius(@PathVariable(name = "tempFarenheit") Double tempFarenheit) {
		return calculadoraTemperatura.farenheitToCelsius(tempFarenheit);
	}
	
	@GetMapping("/ftok/{tempFarenheit}")
	public Double farenheitToKelvin(@PathVariable(name = "tempFarenheit") Double tempFarenheit) {
		return calculadoraTemperatura.farenheitToKelvin(tempFarenheit);
	}
	
	@GetMapping("/ktoc/{tempKelvin}")
	public Double kelvinToCelsius(@PathVariable(name = "tempKelvin") Double tempKelvin) {
		return calculadoraTemperatura.kelvinToCelsius(tempKelvin);
	}
	
	@GetMapping("/ktof/{tempKelvin}")
	public Double kelvinToFarenheit(@PathVariable(name = "tempKelvin") Double tempKelvin) {
		return calculadoraTemperatura.kelvinToFarenheit(tempKelvin);
	}
	
}

File: src/main/kotlin/com/example/temperatura/converter/controller/TemperaturaConverterController.kt
package com.example.temperatura.converter.controller

import com.example.temperatura.converter.interfaces.CalculadoraTemperatura
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(value = ["/converter"])
class TemperaturaConverterController(val calculadoraTemperatura: CalculadoraTemperatura) {

    @GetMapping("/ctof/{tempCelsius}")
    fun celsiusToFareinheit(@PathVariable(name = "tempCelsius") tempCelsius: Double) =
        calculadoraTemperatura.celsiusToFarenheit(tempCelsius)

    @GetMapping("/ctok/{tempCelsius}")
    fun celsiusToKelvin(@PathVariable(name = "tempCelsius") tempCelsius: Double) =
        calculadoraTemperatura.celsiusToKelvin(tempCelsius)

    @GetMapping("/ftoc/{tempFarenheit}")
    fun farenheitToCelsius(@PathVariable(name = "tempFarenheit") tempFarenheit: Double) =
        calculadoraTemperatura.farenheitToCelsius(tempFarenheit)

    @GetMapping("/ftok/{tempFarenheit}")
    fun farenheitToKelvin(@PathVariable(name = "tempFarenheit") tempFarenheit: Double) =
        calculadoraTemperatura.farenheitToKelvin(tempFarenheit)

    @GetMapping("/ktoc/{tempKelvin}")
    fun kelvinToCelsius(@PathVariable(name = "tempKelvin") tempKelvin: Double) =
        calculadoraTemperatura.kelvinToCelsius(tempKelvin)

    @GetMapping("/ktof/{tempKelvin}")
    fun kelvinToFarenheit(@PathVariable(name = "tempKelvin") tempKelvin: Double) =
        calculadoraTemperatura.kelvinToFarenheit(tempKelvin)
}